### ISSUE TITLE
Couse, CourseImporter, WikiEdits, CoursesController need tests

### ISSUE THREAD

**Issue by thenickcox** [WikiEducationFoundation/WikiEduDashboard#282]
Probably the best way to diminish some of the technical debt of [classes rated F on Code Climate](https://codeclimate.com/github/WikiEducationFoundation/WikiEduDashboard/code?q=rating%3AF) is to make an issue for improving them.

The issue is, the classes can't reliably be refactored without tests.

Done when _at least_ minimal tests are written for each of these classes.

Goal should be no F-rated classes in the app, and possibly a 3.0 rating overall, both of which should be attainable.

I'm happy to take this on if you feel that this is worth doing/in the budget, @ragesoss. Shouldn't take terribly long.

**Note:** These can be treated one at a time, with a [campfire rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) or with a concerted effort (i.e., one class per 2-week period, etc.).

**Comment by ragesoss** [comment]
Excellent, thanks Nick!

CourseImporter is only relevant for legacy courses, which will be over after the summer. This should be a very low priority.

As you see, I've started working on WikiEdits, and my refactor brings it up to an A. There is now a minimal test for the 'assignment update' bit that I factored out into a separate class, we also have minimal testing for the API parts of WikiEdits. What remains to be done is substantive unit tests for some of the other entry points, which I haven't found a good way to do, but that's also code that I didn't touch in the refactor.

CoursesController, I'll let you take the lead on. I haven't wrapped my head around the controllers as much as the models and libs, so I look forward to the results of the refactor!
